hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/TriplesResponse.java

```diff
@@ -7,21 +7,24 @@
 
 public interface TriplesResponse extends DataRetrievalResponse<Iterable<Triple>>
 {
-	Iterable<Triple> getTriples();
-
 	/**
-	 * Returns the number of triples that are returned by the {@link #getTriples()}. 
+	 * Returns the number of triples that are returned by the {@link #getResponseData()}.
 	 */
-	default int getSize(){
-		final Iterable<Triple> triples = getTriples();
-		if (triples instanceof Collection) {
-			return ((Collection<Triple>) triples).size();
-		}
-		// Fallback to manual count
-		int count = 0;
-		for ( Iterator<Triple> it = triples.iterator(); it.hasNext(); it.next() ){
-			count++;
+	default int getSize() {
+		try {
+			final Iterable<Triple> triples = getResponseData();
+			if ( triples instanceof Collection ) {
+				return ((Collection<Triple>) triples).size();
```

Comment:
Since one of the newer versions of Java, such explicit type casts are not needed anymore. Instead, the following should work:

```suggestion
			if ( triples instanceof Collection c ) {
				return c.size();
```
That's a general comment which you can apply for all these cases here.

Reply:
Yes, I'll keep it mind, the code often becomes much clearer. I've now updated `FederationAccessManagerBase1.java` and `SolMapsResponse.java` (3e7d7134adf715f44d71fa4f3eeefa015564a87e). But there are many more cases in the code base where we are not taking advantage of this pattern.

Do you think we should create a separate issue for it where only do refactoring using the pattern matching syntax?Or should we just try to keep in mind whenever we update the code?

Reply:
> But there are many more cases in the code base where we are not taking advantage of this pattern.

Indeed. These are from before I knew of this pattern, and from before switching the project to a version of Java in which these patterns are supported.

> Do you think we should create a separate issue for it where we only do refactoring using the pattern matching syntax?
> Or should we just try to keep in mind whenever we update the code?

My strategy so far has been not to change all of the cases in which the patterns can be used but, instead, only make such changes in parts of the code that I am touching for some other reason (and, of course, to use the pattern for new code). But perhaps it is better to have a dedicated PR to implement this change all over the code base instead of having such changes show up now and then within PRs that are actually about something else ;-)